aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -131,23 +134,35 @@ def total_time(self) -> float:
         return (self.time_write_stop or self.time_collect_stop) - self.time_collect_start
 
 
+# should make more generic
+ProgressContext = Callable[..., ContextManager[tqdm]]
+
+
 class ArchiveWriterAbstract(ABC):
     """An abstract interface for AiiDA archive writers."""
 
-    def __init__(self, filename: str, **kwargs: Any):
+    def __init__(self, filename: str, progress_context: Optional[ProgressContext] = None, **kwargs: Any):
         """An archive writer
 
         :param filename: the filename (possibly including the absolute path)
             of the file on which to export.
+        :param progress_context: A context manager for creating and updating a progress bar
+
         """
         # pylint: disable=unused-argument
         self._filename = filename
+        self._progress_context = progress_context or partial(tqdm, bar_format=BAR_FORMAT)
```

Comment:
```suggestion
        self._progress_context = progress_context or partial(tqdm, disable=True)
```

perhaps the default should be to not print anything?

Reply:
right, that's the question of the dummy logger...
I don't have a clear answer at the moment; perhaps we should look around a bit what others have found to work best in such cases.